app/rust/src/parser/message.rs

```diff
@@ -118,7 +118,11 @@ impl<'a> ByteString<'a> {
         });
 
         let mut copy_len = if self.0.len() > MAX_ASCII_LEN {
-            msg[MAX_ASCII_LEN..MAX_ASCII_LEN + suffix.len()].copy_from_slice(&suffix[..]);
+            if let Some(m) = msg.get_mut(MAX_ASCII_LEN..MAX_ASCII_LEN + suffix.len()) {
```

Comment:
`.get_mut(xyz).ok_or(ParserError::xyz)?`

Here the compiler may be smart enough to see that the slice was just `suffix.len()` in length, so the panic should be avoidable... would add a check